crates/fuel-core/src/schema/node_info.rs

```diff
@@ -168,3 +202,24 @@ impl TxPoolStats {
         self.0.total_gas.into()
     }
 }
+
+#[derive(Clone)]
+struct Indexation(HashSet<IndexationKind>);
```

Comment:
I think you could use something like [enumset](https://docs.rs/enumset/latest/enumset/) which allow you to store all the data you need on few bytes and I think will be more than enough for the foreseeable future.

With this new structure you could also change the way you are creating you `HashSet` in `node_info` above and avoid any iteration I think

Reply:
I'm not a big fan of adding a new dependency and rely on new data types just to save a byte or two, that's why I prepare this as a [separate PR](https://github.com/FuelLabs/fuel-core/pull/2601) to this branch. If we want to merge this, I'm ok.

Reply:
I think we don't really need to use the dependency directly but it was an example for an idea. I will try to do it on a branch of top of this one today/tomorrow.

Reply:
Here it is : https://github.com/FuelLabs/fuel-core/pull/2604

The creation of the `Indexation` if a bit more algorithm complex than just : 
```rust
        if read_view.coins_to_spend_indexation_enabled {
            indexation.insert(IndexationKind::CoinsToSpend);
        }
        if read_view.balances_indexation_enabled {
            indexation.insert(IndexationKind::Balances);
        }
        if read_view.asset_metadata_indexation_enabled {
            indexation.insert(IndexationKind::AssetMetadata);
       }
```
but it enforce at compile level that all `InstructionKind` are listed and inserted with a condition. I think that it provide more safety for the codebase and future contributions but up to you :)

Reply:
Approved! :+1: